documents/src/pages/elements/email-field.md

```diff
@@ -283,6 +283,163 @@ emailField?.addEventListener("input", () => {
 });
 ```
 
+### Custom Validation
+
+For advance use cases, apps can control validation and error state of the field manually. Do not set `maxLength` & `minLength` validation constrain and set `pattern` to `''`, then validate the field and update `error` property as needed.
+
+::
+
+```javascript
+::text-field::
+const fullNameField = document.getElementById("full-name");
+const emailField = document.getElementById("email");
+const responseText = document.getElementById("response-text");
+const save = document.getElementById("save");
+
+save.addEventListener("tap", () => {
+  const incompleteError = Boolean(fullNameField.value) !== Boolean(emailField.value);
```

Comment:
```suggestion
  const isValid = Boolean(fullNameField.value) === Boolean(emailField.value);
```
```suggestion
  const isInvalid = Boolean(fullNameField.value) !== Boolean(emailField.value);
```

Reply:
Change to `isIncomplete` as the form could still be invalid if the provided email doesn't match the expected pattern.

Reply:
Let's make it positive ? Something like `isComplete` maybe ?

Reply:
double negative could be confusing, indeed! Changed to `isPartial`